File: 0405-convert-a-number-to-hexadecimal/0405-convert-a-number-to-hexadecimal.py
class Solution:
    def toHex(self, num: int) -> str:
        if num == 0:
            return "0"
        
        num &= 0xFFFFFFFF  
        
        hex_chars = "0123456789abcdef"
        s = ""
        
        while num > 0:
            rem = num % 16
            s += hex_chars[rem]
            num //= 16
        
        return s[::-1]

File: 0405-convert-a-number-to-hexadecimal/0405-convert-a-number-to-hexadecimal.cpp
class Solution {
public:
    string toHex(int num) {
        if (num == 0) return "0";

        unsigned int n = num;      
        string hex_chars = "0123456789abcdef";
        string s = "";

        while (n > 0) {
            s += hex_chars[n % 16];
            n /= 16;
        }

        return string(s.rbegin(), s.rend());
    }
};
